irohad/ametsuchi/impl/postgres_wsv_command.cpp

```diff
@@ -17,335 +17,379 @@
 
 #include "ametsuchi/impl/postgres_wsv_command.hpp"
 
+#include <exception>
 #include <iterator>
 #include <numeric>
 
 namespace iroha {
   namespace ametsuchi {
-
     PostgresWsvCommand::PostgresWsvCommand(pqxx::nontransaction &transaction)
         : transaction_(transaction), log_(logger::log("PostgresWsvCommand")) {}
 
-    bool PostgresWsvCommand::insertRole(const std::string &role_name) {
+    const char *kUnsuccesfulAlloc = "Unable to alloc: {}";
+
+    template <typename T>
+    std::string try_quote(PostgresWsvCommand *that,
+                          std::initializer_list<T> l) noexcept {
       try {
-        transaction_.exec("INSERT INTO role(role_id) VALUES ("
-                          + transaction_.quote(role_name)
-                          + ");");
-      } catch (const std::exception &e) {
-        log_->error(e.what());
-        return false;
+        return try_quote(
+            that,
+            std::accumulate(l.begin(),
+                            l.end(),
+                            std::string(),
+                            [](std::string s, auto next) { return s + next; }));
```

Comment:
pass by ref

Reply:
Not really possible, the first one is rvalue and the second is from the initializer list

Reply:
auto&& s, auto &&next ?